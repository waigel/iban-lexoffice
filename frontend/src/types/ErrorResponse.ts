export interface ErrorResponse {
  code: string;
  params?: string[];
}
