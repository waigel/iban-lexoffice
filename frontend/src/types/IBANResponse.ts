export interface IBANResponse {
  bankName: string;
  countryCode: string;
  countryName: string;
  isInSwiftRegistry: boolean;
  isSepaCountry: boolean;
  bankCode?: string;
  blzRecord: BLZRecord;
}

export interface BLZRecord {
  blz?: string;
  bankName?: string;
  shortBankName?: string;
  zipCode?: string;
  city?: string;
  bic?: string;
}
