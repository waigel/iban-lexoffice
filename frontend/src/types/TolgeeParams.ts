export interface TolgeeParams {
  [key: string]: string;
}
