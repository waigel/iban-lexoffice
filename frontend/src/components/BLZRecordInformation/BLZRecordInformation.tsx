import {useTranslate} from '@tolgee/react';
import React from 'react';
import {BLZRecord} from '../../types/IBANResponse';

export interface BLZRecordInformationProps {
  value?: BLZRecord;
}

export const BLZRecordInformation = ({value}: BLZRecordInformationProps) => {
  const {t} = useTranslate();
  if (!value) {
    return null;
  }

  const className = 'text-gray-300 flex gap-2';
  return (
    <div className="flex flex-col md:pt-6">
      {(value.bankName || value.shortBankName) && (
        <p className={className}>
          <b>{t('BLZ_INFORMATION_LABEL_BANK_NAME')}</b>
          {value.shortBankName ?? value.bankName}
        </p>
      )}

      {value.bic && (
        <p className={className}>
          <b>{t('BLZ_INFORMATION_LABEL_BIC')}</b>
          {value.bic}
        </p>
      )}

      {value.zipCode && value.city && (
        <p className={className}>
          <b>{t('BLZ_INFORMATION_LABEL_BANK_ADDRESS')}</b>
          {value.zipCode} {value.city}
        </p>
      )}
    </div>
  );
};
