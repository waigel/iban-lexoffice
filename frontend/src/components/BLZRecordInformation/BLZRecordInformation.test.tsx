import React from 'react';
import TestRenderer from 'react-test-renderer';
import {BLZRecord} from '../../types/IBANResponse';
import {BLZRecordInformation} from './BLZRecordInformation';

jest.mock('@tolgee/react', () => ({
  useTranslate: () => ({t: jest.fn(key => key), isLoading: false}),
}));

const blzRecord: BLZRecord = {
  blz: '10000000',
  bic: 'GENODEF1M04',
  bankName: 'Deutsche Bank AG',
  shortBankName: 'Deutsche Bank',
  zipCode: '60311',
  city: 'Frankfurt am Main',
};

it('should test the BLZRecordInformation component is rendered correctly', () => {
  const tree = TestRenderer.create(<BLZRecordInformation value={blzRecord} />);
  expect(tree.toJSON()).toMatchSnapshot();
});

it('should display long bankName if shortBankName is not set', () => {
  const blzRecordCopy: BLZRecord = {...blzRecord, shortBankName: undefined};
  const tree = TestRenderer.create(
    <BLZRecordInformation value={blzRecordCopy} />
  );
  expect(tree.toJSON()).toMatchSnapshot();
});

it('should not display bankName if it is not set', () => {
  const blzRecordCopy: BLZRecord = {
    ...blzRecord,
    bankName: undefined,
    shortBankName: undefined,
  };
  const tree = TestRenderer.create(
    <BLZRecordInformation value={blzRecordCopy} />
  );
  expect(tree.toJSON()).toMatchSnapshot();
});

it('should not display bic if it is not set', () => {
  const blzRecordCopy: BLZRecord = {...blzRecord, bic: undefined};
  const tree = TestRenderer.create(
    <BLZRecordInformation value={blzRecordCopy} />
  );
  expect(tree.toJSON()).toMatchSnapshot();
});

it('should not display address if it is not set', () => {
  const blzRecordCopy: BLZRecord = {
    ...blzRecord,
    zipCode: undefined,
    city: undefined,
  };
  const tree = TestRenderer.create(
    <BLZRecordInformation value={blzRecordCopy} />
  );
  expect(tree.toJSON()).toMatchSnapshot();
});
