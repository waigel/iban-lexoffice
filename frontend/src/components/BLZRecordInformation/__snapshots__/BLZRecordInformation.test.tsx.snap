// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display long bankName if shortBankName is not set 1`] = `
<div
  className="flex flex-col md:pt-6"
>
  <p
    className="text-gray-300 flex gap-2"
  >
    <b>
      BLZ_INFORMATION_LABEL_BANK_NAME
    </b>
    Deutsche Bank AG
  </p>
  <p
    className="text-gray-300 flex gap-2"
  >
    <b>
      BLZ_INFORMATION_LABEL_BIC
    </b>
    GENODEF1M04
  </p>
  <p
    className="text-gray-300 flex gap-2"
  >
    <b>
      BLZ_INFORMATION_LABEL_BANK_ADDRESS
    </b>
    60311
     
    Frankfurt am Main
  </p>
</div>
`;

exports[`should not display address if it is not set 1`] = `
<div
  className="flex flex-col md:pt-6"
>
  <p
    className="text-gray-300 flex gap-2"
  >
    <b>
      BLZ_INFORMATION_LABEL_BANK_NAME
    </b>
    Deutsche Bank
  </p>
  <p
    className="text-gray-300 flex gap-2"
  >
    <b>
      BLZ_INFORMATION_LABEL_BIC
    </b>
    GENODEF1M04
  </p>
</div>
`;

exports[`should not display bankName if it is not set 1`] = `
<div
  className="flex flex-col md:pt-6"
>
  <p
    className="text-gray-300 flex gap-2"
  >
    <b>
      BLZ_INFORMATION_LABEL_BIC
    </b>
    GENODEF1M04
  </p>
  <p
    className="text-gray-300 flex gap-2"
  >
    <b>
      BLZ_INFORMATION_LABEL_BANK_ADDRESS
    </b>
    60311
     
    Frankfurt am Main
  </p>
</div>
`;

exports[`should not display bic if it is not set 1`] = `
<div
  className="flex flex-col md:pt-6"
>
  <p
    className="text-gray-300 flex gap-2"
  >
    <b>
      BLZ_INFORMATION_LABEL_BANK_NAME
    </b>
    Deutsche Bank
  </p>
  <p
    className="text-gray-300 flex gap-2"
  >
    <b>
      BLZ_INFORMATION_LABEL_BANK_ADDRESS
    </b>
    60311
     
    Frankfurt am Main
  </p>
</div>
`;

exports[`should test the BLZRecordInformation component is rendered correctly 1`] = `
<div
  className="flex flex-col md:pt-6"
>
  <p
    className="text-gray-300 flex gap-2"
  >
    <b>
      BLZ_INFORMATION_LABEL_BANK_NAME
    </b>
    Deutsche Bank
  </p>
  <p
    className="text-gray-300 flex gap-2"
  >
    <b>
      BLZ_INFORMATION_LABEL_BIC
    </b>
    GENODEF1M04
  </p>
  <p
    className="text-gray-300 flex gap-2"
  >
    <b>
      BLZ_INFORMATION_LABEL_BANK_ADDRESS
    </b>
    60311
     
    Frankfurt am Main
  </p>
</div>
`;
